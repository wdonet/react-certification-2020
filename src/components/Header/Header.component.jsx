import React from 'react';
import { useHistory } from 'react-router';
import Switch from '../Switch';
import { HeaderContainer, Loggin, Menu, SearchInput, SearchLabel } from './Header.styled';
import { DARK_THEME } from '../../utils/constants';

function Header() {
  const history = useHistory();

  function search({ key, target }) {
    if (key === 'Enter') {
      history.push(`/search/${target.value}`);
    }
  }

  return (
    <HeaderContainer>
      <Menu
        blurImage="/images/menu.svg"
        focusImage="/images/menu-focus.svg"
        clickHandler={() => console.log('Menu')}
        alt="Menu"
        name="Menu"
      />
      <SearchLabel htmlFor="search-input">search</SearchLabel>
      <SearchInput
        id="search-input"
        placeholder="Search for videos..."
        onKeyDown={search}
      />
      <Switch
        valueName={DARK_THEME}
        textOn="☽"
        textOff="☼"
        onToggle={(darkTheme) => console.log(darkTheme ? 'Dark' : 'Light')}
      />
      <Loggin
        blurImage="/images/loggin.svg"
        focusImage="/images/loggin-focus.svg"
        clickHandler={() => console.log('Loggin')}
        alt="Loggin"
        name="Loggin"
      />
    </HeaderContainer>
  );
}

export default Header;
