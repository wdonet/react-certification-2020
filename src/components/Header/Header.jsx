import React from 'react';
import { Navbar, Form, Col } from 'react-bootstrap';
import { HiMenu } from 'react-icons/hi';

import { Button } from '../Buttons/Button';
import InputText from '../TextInput/TextInput';
import ProfileImage from '../ProfileImage/ProfileImage';

function Header() {
  return (
    <Navbar bg="light" expand="lg">
      <Col sm={6}>
        <Button>
          <HiMenu />
        </Button>
        <InputText placeholder="Search" />
      </Col>
      <Col sm={{ offset: 3, span: 3 }}>
        <Form inline>
          <Form.Check type="switch" id="custom-switch" label="Dark mode" />
          <ProfileImage src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s88-c-k-c0x00ffffff-no-rj" />
        </Form>
      </Col>
    </Navbar>
  );
}

export default Header;
