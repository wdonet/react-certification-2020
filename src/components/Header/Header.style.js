import styled from "styled-components";


const header = styled.div`
    width: 100%;
    background-color: whitesmoke;
    opacity: 80%;
`

const burgerMenu = styled.div`
    display: inline-block;
    width: 80px;
`

const inputWrapper = styled.div`
    width: 70%;
    display: inline-block;
`

const inputSearch = styled.input`
    width: 250px;
    margin-left: 5%;
    height: 30px;
    border-radius: 8px;
    background: gray;
    opacity: 80%;
    border: none;
    color: white;
    padding: 0px 20px;
    &::-webkit-input-placeholder { 
        color: white; 
        font-size: 15px; 
    } 
`

const toggleWrapper = styled.div`
    position: relative;
    /* justify-content: left; */
    width: 15%;
    display: inline-block;
    padding-left: 3%
`

const toggleBox = styled.label`
    position: absolute;
    top: 13px;
    /* left: 0; */
    width: 42px;
    height: 26px;
    border-radius: 15px;
    background: #bebebe;
    cursor: pointer;
    &::after {
        content: "";
        display: block;
        border-radius: 50%;
        width: 18px;
        height: 18px;
        margin: 3px;
        background: #ffffff;
        box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.2);
        transition: 0.2s;
    }
`

const togglebutton = styled.input`
    opacity: 0;
    z-index: 1;
    border-radius: 15px;
    width: 42px;
    height: 26px;
    &:checked + ${toggleBox} {
        background: #4fbe79;
        &::after {
            content: "";
            display: block;
            border-radius: 50%;
            width: 18px;
            height: 18px;
            margin-left: 21px;
            transition: 0.2s;
        }
    }
`

const darkMode = styled.p`
    display: inline-block;
    font-size: 15px;
    margin-left: 10px;
`
const profile = styled.img`
    width: 50px;
    border-radius: 50%;
    box-shadow: 2px 2px 25px 3px black;
`


const Style = {burgerMenu, header, inputSearch, inputWrapper, toggleWrapper, toggleBox, togglebutton, darkMode, profile}
export default Style

