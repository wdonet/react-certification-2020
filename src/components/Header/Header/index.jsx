import React from 'react';
import Styled from './styled';
import './header.css';

import Logo from '../Logo';
import SearchBar from '../SearchBar';
import Login from '../Login';

const Header = () => {
  return (
    <Styled.Header>
      <Styled.LogoImage>
        <Logo />
      </Styled.LogoImage>
      <Styled.SearchArea>
        <SearchBar />
      </Styled.SearchArea>
      <Styled.Login>
        <Login />
      </Styled.Login>
    </Styled.Header>
  );
};

export default Header;
