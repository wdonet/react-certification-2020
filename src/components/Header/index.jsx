import React from 'react';
import Button from '../Button';
import Toolbar from '../Toolbar';
import IconButton from '../IconButton';
import { Menu } from '../Icons';

import { StyledHeader } from './styled';

const Header = () => {
  return (
    <StyledHeader>
      <Toolbar>
        <Button>
          <IconButton>
            <Menu width="24px" color="white" />
          </IconButton>
        </Button>
      </Toolbar>
    </StyledHeader>
  );
};

export default Header;
