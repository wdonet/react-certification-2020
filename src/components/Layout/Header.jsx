import React from 'react';
import styled from 'styled-components';
import { TextField, IconWrapper, Avatar, Switch } from '../Base/index';
import hamburger from '../../assets/icons/hamburguer.png';
import defaultUser from '../../assets/icons/default_user.jpg';

const StyledNavbar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 64px;
  background-color: #00695c;
  overflow: hidden;
  position: fixed;
  top: 0;
`;

const StyledSection = styled.div`
  display: flex;
  align-items: center;
`;

const Navbar = () => {
  return (
    <StyledNavbar role="toolbar" data-testid="navbar">
      <StyledSection>
        <div>
          <IconWrapper role="button" src={hamburger} alt="hamburguer" />
        </div>
        <div data-testid="search-input">
          <TextField role="search"/>
        </div>
      </StyledSection>
      <StyledSection>
        <div data-testid="theme-mode-switch">
          <Switch />
        </div>
        <div data-testid="user-avatar">
          <Avatar role="button" alt="profile" src={defaultUser} />
        </div>
      </StyledSection>
    </StyledNavbar>
  );
};

export default Navbar;
