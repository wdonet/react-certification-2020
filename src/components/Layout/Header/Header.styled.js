import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

const Col = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 10px;
`;

const TitleLink = styled(NavLink)`
  text-decoration: none;
  color: black;
  ${({ theme }) => (theme.color ? `color: ${theme.color}` : '')};
`;

const HeaderContainer = styled.div`
  display: flex;
  flex-direction: row;
  height: 70px;
  align-items: center;
  padding: 5px;
  border-bottom: 3px solid darkgray;
  background-color: lightgray;
  ${({ theme }) => (theme.primary ? `background-color: ${theme.primary}` : '')};
  ${({ theme }) => (theme.color ? `color: ${theme.color}` : '')};
`;

const Search = styled.input`
  background-color: lightyellow;
  border: 0;
  display: flex;
  height: 1.8rem;
  padding: 5px;
`;

const Space = styled.div`
  display: flex;
  flex: 1;
`;

const Switch = styled.span`
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const User = styled.span`
  height: 40px;
  background-color: springgreen;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export { Col, TitleLink, HeaderContainer, Search, Space, Switch, User };
