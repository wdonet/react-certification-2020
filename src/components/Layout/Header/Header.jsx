import React from 'react';
import { useHistory } from 'react-router-dom';
import { useAuth } from '../../../providers/Auth';
import { useYouTube } from '../../YouTube/YouTubeProvider';
import { Col, TitleLink, HeaderContainer, Search, Space, Switch } from './Header.styled';

const Header = () => {
  const { state, dispatch } = useYouTube();
  const { search = '', theme } = state;

  const { authenticated, logout } = useAuth();
  const history = useHistory();

  const setSearch = (event) => {
    dispatch({ type: 'search', payload: event.target.value });
  };

  return (
    <HeaderContainer>
      <TitleLink to="/">
        <Col>React Bootcamp</Col>
      </TitleLink>
      <Col>
        <Search placeholder="Search" value={search} onChange={setSearch} />
      </Col>
      <Space />
      <TitleLink to="/">
        <Col>Videos</Col>
      </TitleLink>
      <TitleLink to="/favorites">
        <Col>Favorites</Col>
      </TitleLink>
      <Col>
        <Switch
          onClick={() => {
            dispatch({ type: 'switchTheme' });
          }}
        >
          {theme === 'dark' ? 'Default' : 'Dark'}
        </Switch>
      </Col>

      <Col>
        <button
          type="button"
          onClick={() => {
            if (authenticated) {
              logout();
            } else history.push('/login');
          }}
        >
          {authenticated ? 'Logout' : 'Login'}
        </button>
      </Col>
    </HeaderContainer>
  );
};

export default Header;
