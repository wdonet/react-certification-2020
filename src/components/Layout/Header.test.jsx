import React from 'react';
import 'jest-styled-components';
import { getByRole, render } from '@testing-library/react';
import Header from './Header';

const build = (Component = <Header />) => {
  const { container } = render(Component);
  return {
    container,
    HamburguerIcon: () => getByRole(container, "button", { name: "hamburguer" }),
    SearchInput: () => getByRole(container, 'search'),
    ThemeModeSwitch: () => getByRole(container, 'switch'),
    UserAvatar: () => getByRole(container, "button", { name: "profile" }),
  };
};

describe('Header', () => {
  it('renders', () => {
    const { container } = build();
    expect(container).toMatchSnapshot();
  });

  it('displays its default content', () => {
    const { HamburguerIcon, SearchInput, ThemeModeSwitch, UserAvatar } = build();
    expect(HamburguerIcon()).toBeInTheDocument();
    expect(SearchInput()).toBeInTheDocument();
    expect(ThemeModeSwitch()).toBeInTheDocument();
    expect(UserAvatar()).toBeInTheDocument();
  });
});

describe('Header styles', () => {
  it('applies default styling', () => {
    const { firstChild } = build().container;
    expect(firstChild).toHaveStyle('display: flex');
    expect(firstChild).toHaveStyle('align-items: center');
    expect(firstChild).toHaveStyle('justify-content: space-between');
    expect(firstChild).toHaveStyle('width: 100%');
    expect(firstChild).toHaveStyle('height: 64px');
    expect(firstChild).toHaveStyle('background-color: #00695c');
    expect(firstChild).toHaveStyle('overflow: hidden');
    expect(firstChild).toHaveStyle('position: fixed');
    expect(firstChild).toHaveStyle('top: 0');
  });
});
