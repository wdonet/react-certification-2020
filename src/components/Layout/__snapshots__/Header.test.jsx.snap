// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div>
  <div
    class="sc-ifAKCX fOhgAo"
    data-testid="navbar"
    role="toolbar"
  >
    <div
      class="sc-EHOje hecEEV"
    >
      <div>
        <img
          alt="hamburguer"
          class="sc-bwzfXH eRyTNY"
          role="button"
          src="hamburguer.png"
        />
      </div>
      <div
        data-testid="search-input"
      >
        <input
          class="sc-bdVaJa kHpFHH"
          role="search"
        />
      </div>
    </div>
    <div
      class="sc-EHOje hecEEV"
    >
      <div
        data-testid="theme-mode-switch"
      >
        <input
          class="sc-bxivhb bUZGRr"
          role="switch"
          type="checkbox"
        />
      </div>
      <div
        data-testid="user-avatar"
      >
        <img
          alt="profile"
          class="sc-htpNat ihTYrT"
          role="button"
          src="default_user.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
